pkg/git/command_darwin.go

```diff
@@ -0,0 +1,27 @@
+// +build darwin
+
+// Copyright 2020 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package git
+
+import (
+	"os/exec"
+	"strings"
+)
+
+func copyCommand(src, dst string) *exec.Cmd {
+	src = strings.TrimSuffix(src, "/")
+	return exec.Command("cp", "-rf", src+"/.", dst)
```

Comment:
nits, does the `cp` command not work without the `.` at the end of src path? 👀

Reply:
Me too. Not sure why do we need this change?

Reply:
MacOS's `cp -r` puts the directory you're copying into the destination directory.

```
❯ ls -a /var/folders/dn/4wqz0bq15x5_5zpxf1grd455pmlkzw/T/srcdir/
./             ../            .git/          Dockerfile     client/        client2/       go.mod         go.sum         main.go

❯ cp -rf /var/folders/dn/4wqz0bq15x5_5zpxf1grd455pmlkzw/T/srcdir /var/folders/dn/4wqz0bq15x5_5zpxf1grd455pmlkzw/T/dstdir

❯ ls -a /var/folders/dn/4wqz0bq15x5_5zpxf1grd455pmlkzw/T/dstdir
./                      ../                     srcdir/
```

We can copy only the files under the srcdir by keeping the trailing slash for srcdir:
```
❯ cp -rf /var/folders/dn/4wqz0bq15x5_5zpxf1grd455pmlkzw/T/srcdir/ /var/folders/dn/4wqz0bq15x5_5zpxf1grd455pmlkzw/T/dstdir

❯ ls -a /var/folders/dn/4wqz0bq15x5_5zpxf1grd455pmlkzw/T/dstdir
./             ../            .git/          Dockerfile     client/        client2/       go.mod         go.sum         main.go
```

Reply:
Actually the trailing `.` is unneeded.

Reply:
I see, then this part is to ensure the last character of `src` path is `/` if I guess right? 🤔 
besides, I checked on my macOS environment, `src/path/` and `src/path//` have the same effect to the `cp` command 😅 I think we could simplify this logic here as `just add the '/' character` at the end of `src` path. How do you think about that?

Reply:
Yup, I'll do so after gaining everyone's agreements.

Reply:
I checked on my Linux environment, looks like the `cp` Linux command has the same behavior as macOS command, which means `cp -r` will copy the entry directory as a subdirectory of the `dist` path. If you want to just copy all subdirectories and files of the `src` directory to the dist path, you should add `*` to the end of the `src` path in case of using the Linux `cp` command.

Reply:
Is it true? Seems like I need to look into it more.